src/ztex_phpass.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * This software is Copyright (c) 2018 Denis Burykin
+ * [denis_burykin yahoo com], [denis-burykin2014 yandex ru]
+ * and it is hereby released to the general public under the following terms:
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * This file is part of John the Ripper password cracker,
+ * Copyright (c) 1996-2001,2008,2010-2013,2015 by Solar Designer
+ *
+ * This file is part of John the Ripper password cracker,
+ * Copyright (c) 2012 magnum / JimF
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * There's ABSOLUTELY NO WARRANTY, express or implied.
+ *
+ */
+#include <stdio.h>
+#include <string.h>
+
+#include "arch.h"
+#include "common.h"
+#include "formats.h"
+#include "memory.h"
+#include "config.h"
+#include "options.h"
+#include "phpass_common.h"
+
+#include "ztex/device_bitstream.h"
+#include "ztex/device_format.h"
+#include "ztex/task.h"
+#include "ztex/pkt_comm/cmp_config.h"
+
+
+#define FORMAT_LABEL		"phpass-ztex"
+#define ALGORITHM_NAME		"phpass ($P$ or $H$)"
+
+#define FORMAT_NAME		""
+
+#define BENCHMARK_COMMENT	""
+
+// Partial hash in the database - differs from CPU/GPU implementations
+#undef	BINARY_SIZE
+#define	BINARY_SIZE		4
+
+// Actual salt size is 9
+#define	SALT_SIZE		8
+#define	SALT_ALIGN		1
+
+
+static struct device_bitstream bitstream = {
+	// bitstream ID (check is performed by querying operating bitstream)
+	0xd5c0,
+	"$JOHN/ztex/ztex115y_md5crypt.bit",
+	// parameters for high-speed packet communication (pkt_comm)
+	{ 2, 14336, 4094 },
+	// computing performance estimation (in candidates per interval)
+	// (keys * mask_num_cand)/crypt_all_interval per jtr_device.
+	1,		// keys/fpga for crypt_all(): set by init()
+	4096,	// keys/fpga for self-test
+	1536 * 1024,	// Would be 48 MB of USB traffic on 32-byte keys
+	512,		// Max. number of entries in onboard comparator.
+	384,		// Min. number of keys for effective device utilization
+	1, { 135 },	// Programmable clocks
```

Comment:
Should probably use same default frequency as we have in the comment in `john.conf`, currently 180.

Reply:
I'm unsure in a _scenario_ where a user decides to set frequency below 180.
This number 135, hardcoded in ```struct device_bitstream``` is used when resetting the FPGA with Global Set Reset.
Then frequency from ```john.conf``` applies. If a user decides, lets say 160 or 170 then one can be sure the FPGA will never run at higher frequency.

Reply:
Oh, I thought the frequency from `john.conf` always overrides this one, and conversely this one is always used when there's no frequency specified in `john.conf`. Is this not the case? Can/should we make it the case? Any other behavior feels non-intuitive.